src/modules/cmdpal/ext/Microsoft.CmdPal.Ext.Bookmark/Services/BookmarkResolver.cs

```diff
@@ -0,0 +1,548 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using ManagedCommon;
+using Microsoft.CmdPal.Ext.Bookmarks.Helpers;
+
+namespace Microsoft.CmdPal.Ext.Bookmarks.Services;
+
+internal sealed partial class BookmarkResolver : IBookmarkResolver
+{
+    private readonly IPlaceholderParser _placeholderParser;
+
+    private const string UriSchemeShell = "shell";
+
+    public BookmarkResolver(IPlaceholderParser placeholderParser)
+    {
+        ArgumentNullException.ThrowIfNull(placeholderParser);
+        _placeholderParser = placeholderParser;
+    }
+
+    public async Task<(bool Success, Classification Result)> TryClassifyAsync(
+        string? input,
+        CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            var result = await Task.Run(
+                () => TryClassify(input, out var classification)
+                    ? classification
+                    : Classification.Unknown(input ?? string.Empty),
+                cancellationToken);
```

Comment:
this `cancellationToken` isn't actually passed down into `ClassifyCore`, so we're going to have to do all the filesystem lookups even if we were cancelled